### PR TITLE
Support navbar images

### PR DESCRIPTION
**Pull request by josh146**
**Context:** Adds support for images in the navigation bar.

**Description of the Change:** For the `navbar_left_links` configuration option, the ``"img"`` key can be used to specify an image for the navbar link, alongside ``"img_width"`` to specify the width of the image.

**Benefits:** As above.

**Possible Drawbacks:** n/a

**Related GitHub Issues:** n/a

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,8 @@
-## Release 0.4.0 (development release)
+## Release 0.3.1 (development release)
+
+### Improvements
+
+* Added support for images in navbar links. [#](https://github.com/XanaduAI/xanadu-sphinx-theme/pull/)
```
Comment (josh146) [inline .github/CHANGELOG.md:5]:
```suggestion
* Added support for images in navbar links. [#21](https://github.com/XanaduAI/xanadu-sphinx-theme/pull/21)
```

File: .github/CHANGELOG.md [hunk 2]
```diff
@@ -10,9 +14,9 @@
 
 This release contains contributions from (in alphabetical order):
 
-[Josh Izaac](https://github.com/josh146)
+[Josh Izaac](https://github.com/josh146).
 
-# Release 0.3.0
+## Release 0.3.0
```
Comment (Mandrenkov) [inline .github/CHANGELOG.md:19]: Shouldn't this be
```suggestion
## Release 0.3.0 (current release)
```

File: doc/conf.py [hunk 3]
```diff
@@ -104,7 +104,10 @@
     # "navbar_home_link": "https://pennylane.ai",
     "navbar_left_links": [
         {
+            # revert this before merging
```
Comment (Mandrenkov) [inline doc/conf.py:107]: Let's do that!

File: xanadu_sphinx_theme/header.html [hunk 4]
```diff
@@ -10,7 +10,15 @@ <h1 id="title">{{ theme_navbar_name }}</h1>
   {% endif %}
 {%- endmacro %}
 
-{% macro hover_toc(name, href, class="", active=False) %}
+{% macro render_navbar_link(name, img=None, width=None) %}
+  {% if img %}
+    <img src="{{ pathto(img, 1) }}" {% if width %}width="{{ width }}" {% endif %} /> {{ name}}
```
Comment (Mandrenkov) [inline xanadu_sphinx_theme/header.html:15]:
```suggestion
    <img src="{{ pathto(img, 1) }}" {% if width %} width="{{ width }}" {% endif %} /> {{ name }}

```

**Review by Mandrenkov — APPROVED** [review]
Looks good to me! 🖼️

File: .github/CHANGELOG.md (hunk 2 above)
Comment (josh146) [reply .github/CHANGELOG.md:19]: Our changelog conventions always confuse me 😆

File: doc/conf.py [hunk 5]
```diff
@@ -104,7 +104,10 @@
...
             "name": "Blog",
+            "img": "_static/teleport.png",
+            "img_width": "30px",
```
Comment (josh146) [inline doc/conf.py:110]:
```suggestion
            "name": "Blog",
```

**Pull request merged**